evennia/utils/tests/test_utils.py

```diff
@@ -292,3 +295,191 @@ def test_byte_string(self):
         byte_str = utils.to_bytes(self.example_str)
         result = utils.latinify(byte_str)
         self.assertEqual(result, self.expected_output)
+
+
+_TASK_HANDLER = None
+
+
+def dummy_func(obj):
+    """
+    Used in TestDelay.
+
+    A function that:
+        can be serialized
+        uses no memory references
+        uses evennia objects
+    """
+    # get a reference of object
+    from evennia.objects.models import ObjectDB
+    obj = ObjectDB.objects.object_search(obj)
+    obj = obj[0]
+    # make changes to object
+    obj.ndb.dummy_var = 'dummy_func ran'
+    return True
+
+
+class TestDelay(EvenniaTest):
+    """
+    Test utils.delay.
+    """
+
+    def test_delay(self):
```

Comment:
Wow ... this is ... very hard to read :sweat_smile: We have some existing tests in Evennia that is structured like a wall of text like this and we are trying to get away from that :wink:. 

I would suggest splitting this into several methods instead, using a `setUp` (and optional `tearDown` method) to set up the taskhandler when testing one specific thing. That way you can run individual tests and it becomes much easier to see which is failing (and you can also have proper doc strings) with less risk of 'contamination' between the tests (each unit test is in principle meant to be stand-alone from others). 
It's still okay to have multiple asserts per test etc, but try to separate the tests into what is being tested.

Reply:
Done with https://github.com/evennia/evennia/pull/2374/commits/cf8052b816c0011330ead39c85e871c85918a0e4